### DISCUSSION TITLE
about edit tool

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#7413] [Q&A]
I'm using the edit tool(SampleMoveElementsTool) to try to move the element. On the desktop App, however, the mouse didn't change to the movable state after I clicked the button.

![pic](https://github.com/user-attachments/assets/1d968fc1-9473-488a-837b-b09a2fd44575)

export class SampleMoveElementsTool extends TransformElementsTool {
    public static override toolId = "SampleMoveElements";
    public static override iconSpec = "icon-move";
    protected calculateTransform(ev: BeButtonEvent): Transform | undefined {

        let mm = (this.anchorPoint ? Transform.createTranslation(ev.point.minus(this.anchorPoint)) : undefined);

        console.log("mm");
        console.log(mm);

        return mm;
        // return Transform.createTranslationXYZ(1,1,1);
    }
    public async onRestartTool(): Promise<void> {
        const tool = new SampleMoveElementsTool();
        if (!await tool.run()) return this.exitTool();
    }
}

the button click this code:
  await (new SampleMoveElementsTool()).run();

**Comment by GytisCepk** [comment]
It sounds like tool fails to start. There could be a couple of reasons for that:
- Make sure to register `SampleMoveElementsTool` during your application initialization with `SampleMoveElementsTool.register()`.
- Your iModel might be opened as read-only. How are you opening connection to your iModel? Did you use Viewer template to start your application or are you opening connection yourself?

Also, if you could share any errors or warnings in the console, that might help us to narrow down the reason.
